test/io/testxyzreader.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Unit tests for XYZ reader."""
+
+import os
+import unittest
+
+import numpy as np
+
+import cclib
+
+
+__filedir__ = os.path.dirname(__file__)
+__filepath__ = os.path.realpath(__filedir__)
+__datadir__ = os.path.join(__filepath__, "..", "..")
+
+
+class XYZReaderTest(unittest.TestCase):
+
+    def test_attributes_single(self):
+        """Is an XYZ file with a single geometry read into a ccData properly?
+        """
+        fpath = os.path.join(__datadir__, "test/bridge/uracil.xyz")
+        assert os.path.exists(fpath)
+        xyz = cclib.io.xyzreader.XYZ(fpath)
+        data = xyz.parse()
+
+        attrs = ('natom', 'atomnos', 'atomcoords')
+        for attr in attrs:
+            self.assertTrue(hasattr(data, attr))
+
+        self.assertEqual(data.natom, 12)
+
+        atomnos = np.array([7, 6, 1, 8, 7, 6, 8, 6, 6, 1, 1, 1], dtype=int)
+
+        self.assertEqual(data.atomnos.shape, atomnos.shape)
+        self.assertTrue(np.all(np.equal(data.atomnos, atomnos)))
+
+        atomcoords = np.array([[
```

Comment:
Can the test be formulated in a way that doesn't require replicating the data?

Reply:
Not if you want to test for explicit correctness, no.

Reply:
Well, what if we wrote it back out and made sure the resulting file was the same?

Reply:
I started to write a roundtrip test, and I still think it's important to have, but I don't know that it solves this problem. The reason is because we need to fix the precision in the output file, and there are many reasons why something like the MD5 of the input and output files would differ. In that case, you could compare the generated output to a reference generated output.

Is that what you're referring to?

Reply:
The numbers in that example don't look weird to get mangled due to rounding or whatever. I wouldn't compare MD5's myself, just the exact strings, but I think it can be done in many ways. I do think not having the data in the actual test would be better, though. If you think it's not worth the effort at this time, that's OK with me as well.